decoders/hdc2080/pd.py

```diff
@@ -0,0 +1,343 @@
+    0x0E: "Soft reset and interrupt configuration",
+    0x0F: "Measurement configuration",
+    0xFC: "Manufacturer ID low",
+    0xFD: "Manufacturer ID high",
+    0xFE: "Device ID low",
+    0xFF: "Device ID high",
+}
+
+regToShort = {
+    0x00: "T LSB",
+    0x01: "T MSB",
+    0x02: "H LSB",
+    0x03: "H MSB",
+    0x04: "Interrupt/drdy",
+    0x05: "T max",
+    0x06: "H max",
+    0x07: "Interrupt enable",
+    0x08: "Temp offset adjust",
+    0x09: "Hum offset adjust",
+    0x0A: "Temp THR LSB",
+    0x0B: "Temp THR MSB",
+    0x0C: "RH THR LSB",
+    0x0D: "RH THR MSB",
+    0x0E: "Reset and DRDY/Int conf",
+    0x0F: "Measurement conf",
+    0xFC: "Man ID low",
+    0xFD: "Man ID high",
+    0xFE: "Dev ID low",
+    0xFF: "Dev ID high",
+}
+
+STATE_IDLE = 1
+STATE_GET_SLAVE_ADDR = 2
+STATE_WAIT_WRITE_ADDR = 3
+STATE_READ_ADDR = 4
+STATE_WRITE_ADDR = 5
+
+ROW_TEMP = 0
+ROW_HUMID = 1
+ROW_WARN = 2
+ROW_READ_PTR = 3
+
+class Decoder(srd.Decoder):
+    api_version = 3
+    id = 'hdc2080'
+    name = 'HDC2080'
+    longname = 'TI HDC2080'
+    desc = 'HDC2080 Low-Power Humidity and Temperature Digital Sensor'
+    license = 'gplv2+'
+    inputs = ['i2c']
+    outputs = []
+    tags = ['IC', 'Sensor']
+    annotations = (
+        ('temperature', 'Temperature / °C'),
+        ('humidity', 'Humidity / RH%'),
+        ('warning', 'Warning'),
+        ('rptr', 'Read Pointer'),
+    )
+    annotation_rows = (
+        ('temp', 'Temperature', (0,)),
+        ('humidity', 'Humitidy', (1,)),
+        ('warning', 'Warnings', (2,)),
+        ('readptrs', 'Read Pointers', (3,)),
+    )
+
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self.state = STATE_IDLE
+        self.reg = 0x00
+
+        self.ss_addr = None
+
+        # how many consecutive bytes
+        # we have read or written in a row
+        self.bytes_num = 0
+
+        self.resetTemp()
+        self.resetHumid()
+
+    def resetTemp(self):
+        self.ss_temp_lsb = None
+        self.ss_temp_msb = None
+
+        self.temp_lsb = None
+        self.temp_msb = None
+
+    def resetHumid(self):
+        self.ss_humid_lsb = None
+        self.ss_humid_msb = None
+
+        self.humid_lsb = None
+        self.humid_msb = None
+
+    def start(self):
+        self.out_ann = self.register(srd.OUTPUT_ANN)
+
+    def getRegisterNameLong(self, reg):
+        try:
+            return regToLong[reg]
+        except:
+            return "Unknown"
+
+    def getRegisterNameShort(self, reg):
+        try:
+            return regToShort[reg]
+        except:
+            return "Unk"
+
+    def warn_upon_invalid_slave(self, addr):
+        # HDC2080 devices have a 7-bit I2C slave address, where the 6 MSBs are
+        # fixed to 100000, and the 1 LSBs can be configured.
+        # Thus, valid slave addresses (100000x) range from 0x40 to 0x41
+        if addr not in [0x40, 0x41]:
+            s = 'Warning: I²C slave 0x%02x not an HDC2080 sensor.'
```

Comment:
does this get printed if there is unrelated traffic on the i2c bus ? e.g. talking to an eeprom at the same time should not cause this PD to spam warnings on every transaction